kernel/comps/framebuffer/src/framebuffer.rs

```diff
@@ -132,6 +192,120 @@ impl FrameBuffer {
         let frame = alloc::vec![0u8; self.io_mem().size()];
         self.write_bytes_at(0, &frame).unwrap();
     }
+
+    /// Sets a single color register in the pseudo palette
+    ///
+    /// For efifb devices, hardware colormap is not supported, so we maintain a
+    /// pseudo palette for software emulation. This function implements
+    /// software colormap emulation similar to Linux's efifb_setcolreg.
+    pub fn set_color_regs(
+        &self,
+        regno: u32,
+        red: u16,
+        green: u16,
+        blue: u16,
+        _transp: u16,
+        cmap_len: u32,
+    ) -> Result<i32> {
+        if regno >= cmap_len {
+            return Ok(1); // Return 1 for invalid regno, matching Linux behavior
```

Comment:
You ignore the error code here:
```rust
       let _ = self.framebuffer.set_color_regs(
```

So, there is nothing that matches the Linux behavior. As I've said before, it isn't reasonable to match the Linux code at the implementation level, and that wasn't our goal.